rest/slurmrestapi.py

```diff
@@ -503,5 +515,49 @@ def proxy():
                            url_root=request.url_root,
                            masters=masters)
 
+
+def check_private_data_for_entity(user, entity):
+    """
+    Return true if the entity is one of the attribute defined previously in
+    Private Data settings.
+    """
+    onlyUsersEntities = False
+    currentUser = user
+
+    if auth_enabled:
+        # Fetch the attributs of private_data
+        config = pyslurm.config().get()
+        private_data = config["private_data_list"]
+
+        if (private_data and entity in private_data and
+                currentUser.role != 'admin'):
+            onlyUsersEntities = True
+    return onlyUsersEntities
+
+
+def filter_entities(entity, entitiesList):
+    onlyUsersEntities = False
+    if auth_enabled:
+        # If auth_enabled is true, getting the current user becomes
+        # possible because authentification_verify decorator has been
+        # already checked.
+        currentUser = get_current_user()
+        onlyUsersEntities = check_private_data_for_entity(
+            currentUser, entity)
+
+    if onlyUsersEntities:
+        # if private data is applied and the jobs' owner is different from
+        # current user, then the jobs will not be added to the job list to
+        # show if auth disabled, onlyUsersJobs becomes always False and all
+        # the jobs are added to the job list to show
+
+        return dict((k, v) for k, v in entitiesList.iteritems()
```

Comment:
Please fix the indentation here:
```
C:556, 0: Wrong continued indentation.
                        v['users']) or (entity == 'jobs' and
                        ^| (bad-continuation)
C:557, 0: Wrong continued indentation.
                        currentUser.login == v['login'])))
                        ^               | (bad-continuation)
```

The readability of the of this nested logic is not that great. I think it would be relevantly replaced by an additional if statement before the return and its iteration...